### PR TITLE
[GPU] Add dynamic functional tests for shapeof op

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *Add dynamic functional tests for shape_of op*
 - *...*

### CODE REVIEW

File: src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+TEST_P(ShapeOfLayerGPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+}
+
+namespace {
+
+const std::vector<ElementType> netPrecisions = {
+        // ElementType::f32,
```
Comment (ahnyoung-paul) [inline src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp:83]: How about deleting it if you don't use it?

File: src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        // ElementType::bf16,
+        ElementType::i32,
+        // ElementType::i8
```
Comment (yeonbok) [inline src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp:86]: Could you remove comments if not used?

File: src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp (hunk 1 above)
Comment (kelvinchoi-intel) [reply src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp]: done

File: src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp (hunk 2 above)
Comment (kelvinchoi-intel) [reply src/tests/functional/plugin/gpu/single_layer_tests/dynamic/shapeof.cpp]: done

**Review by ahnyoung-paul — APPROVED** [review]

**Review by yeonbok — APPROVED** [review]

**Pull request merged**